lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2576,6 +2576,119 @@ static Type validateParameterType(ParamDecl *decl) {
   return Ty;
 }
 
+llvm::Optional<LifetimeDependenceInfo> validateLifetimeDependenceInfo(
+    LifetimeDependentReturnTypeRepr *lifetimeDependentRepr, Decl *decl,
+    bool allowIndex) {
+  auto *afd = cast<AbstractFunctionDecl>(decl);
```

Comment:
I don't remember whether `OnFunc` actually precludes this being called in invalid positions or not but it be good to add a few tests where attribute is placed on i.e. return of a subscript or an accessor to make sure that this doesn't crash.

Reply:
Good point. I added new diagnostics when the lifetime dependence specifiers are not in function result position. Added a test for subscript.